src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,601 @@
+            while (itr.hasNext()) {
+                Map.Entry<Long,Long> entry = itr.next();
+                if ((nowMs - entry.getValue() > kStreamTimeOutMs)) {
+                    itr.remove();
+                }
+            }
+            if (ssrcs_.isEmpty()) {
+                // We can't update the estimate if we don't have any active streams.
+                interArrival = new InterArrival((kTimestampGroupLengthMs
+                        << kInterArrivalShift) / 1000,
+                        kTimestampToMs, true);
+                estimator = new OveruseEstimator(new OverUseDetectorOptions());
+                // We deliberately don't reset the first_packet_time_ms_
+                // here for now since we only probe for bandwidth in the
+                // beginning of a call right now.
+            }
+        }
+    }
+
+    void OnRttUpdate(long avg_rtt_ms,
+                     long max_rtt_ms)
+    {
+        synchronized (critSect) {
+            remoteRate.setRtt(avg_rtt_ms);
+        }
+    }
+
+    void removeStream(long ssrc)
+    {
+        synchronized (critSect)
+        {
+            ssrcs_.remove(ssrc);
+        }
+    }
+
+    @Override
+    public long getLatestEstimate() {
+
+        synchronized (critSect)
+        {
+            long bitrateBps;
+            if (!remoteRate.isValidEstimate()) {
+                return -1;
+            }
+            if (ssrcs_.isEmpty()) {
+                bitrateBps = 0;
+            } else {
+                bitrateBps = remoteRate.getLatestEstimate();
+            }
+            return bitrateBps;
+        }
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTP packets
+     */
+    @Override
+    public PacketTransformer getRTPTransformer() {
+        return this;
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTCP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTCP packets
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer() {
+        return null;
+    }
+
+    /**
+     * Returns the estimated payload bitrate in bits per second if a valid
+     * estimate exists; otherwise, <tt>-1</tt>.
+     *
+     * @return the estimated payload bitrate in bits per seconds if a valid
+     * estimate exists; otherwise, <tt>-1</tt>
+     */
+
+    @Override
+    public Collection<Integer> getSsrcs() {
+
+        synchronized (critSect)
+        {
+            ssrcs = new ArrayList<>();
+            for(Long ssrcValue : ssrcs_.keySet()){
+                Number value = ssrcValue;
+                ssrcs.add(value.intValue());
+            }
+            return ssrcs;
+        }
+
+    }
+
+
+    /**
+     * Removes all data for <tt>ssrc</tt>.
+     *
+     * @param ssrc
+     */
+    @Override
+    public void removeStream(int ssrc)
+    {
+        synchronized (critSect) {
+            try {
+                ssrcs_.remove(ssrc);
```

Comment:
You need to handle the conversion from int to long here: ssrcs_.remove(ssrc & 0xFFFF_FFFFL)